src/Umbraco.Web.Common/ModelsBuilder/InMemoryModelFactory.cs

```diff
@@ -68,7 +68,7 @@ internal class InMemoryModelFactory : IAutoPublishedModelFactory, IRegisteredObj
             _umbracoServices = umbracoServices;
             _profilingLogger = profilingLogger;
             _logger = logger;
-            _config = config.Value;
+            _config = config.CurrentValue;
```

Comment:
`InMemoryModelFactory` is a singleton, so the config won't update unless we hook into the event

Reply:
We should actually still do that in the ModelsGenerationError.cs

Reply:
Yes, but that `ModelsBuilderSettings` is also used within `InMemoryModelFactory` for instance on line 138:

```C#
public bool Enabled => _config.ModelsMode == ModelsMode.InMemoryAuto;
```